### PR TITLE
JDBC HW 03

### PR DESCRIPTION
**Pull request by Max-Pochepets**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,27 @@
...
+    - boolean delete(Long id);
+- Create Dao and Service layers for `Driver` and theirs implementations.
+### DriverService methods:
+    - Driver create(Driver driver);
+    - Driver get(Long id);
+    - List<Driver> getAll();
+    - Driver update(Driver driver);
+    - boolean delete(Long id);
+
+- Test your code in `main`.
```
Comment (KseniiaMakarova) [inline README.md:25]: let's add that they should put `@Service` annotation above the service implementations

and that they should test only service methods in Main, not dao

File: src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+        } catch (SQLException throwable) {
+            throw new DataProcessingException("Couldn't create manufacturer. " + manufacturer + " ",
+                    throwable);
+        }
+    }
+
+    @Override
+    public Optional<Manufacturer> get(Long id) {
+        String query = "SELECT manufacturer_id, manufacturer_name, "
+                + "manufacturer_country FROM manufacturers"
```
Comment (KseniiaMakarova) [inline src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java:40]:
```suggestion
        String query = "SELECT * FROM manufacturers"
```

File: src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+        String country = resultSet.getString("manufacturer_country");
+        Manufacturer manufacturer = new Manufacturer(name, country);
+        manufacturer.setId(newId);
+        return manufacturer;
+    }
+
+    private PreparedStatement setUpdate(PreparedStatement statement,
+                                        Manufacturer manufacturer) throws SQLException {
+        statement.setString(1, manufacturer.getName());
+        statement.setString(2, manufacturer.getCountry());
```
Comment (KseniiaMakarova) [inline src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java:115]: мб лучше не выносить это в приватный метод?

File: README.md [hunk 4]
```diff
@@ -1 +1,27 @@
...
+}
+```
+- Create `ManufacturerService` and its implementation. Annotation `@Service` has already been given to you.
+### ManufacturerService methods:
+    - Manufacturer create(Manufacturer manufacturer);
+    - Manufacturer get(Long id);
+    - List<Manufacturer> getAll();
+    - Manufacturer update(Manufacturer manufacturer);
+    - boolean delete(Long id);
+- Create Dao and Service layers for `Driver` and theirs implementations.
```
Comment (KseniiaMakarova) [inline README.md:17]:
```suggestion
- Create Dao and Service layers for `Driver` and their implementations.
```

File: README.md [hunk 5]
```diff
@@ -1 +1,29 @@
...
+    - Driver get(Long id);
+    - List<Driver> getAll();
+    - Driver update(Driver driver);
+    - boolean delete(Long id);
+
+- Test your code in `main`.
+- Main should only contain invocation of `Service` methods, no Dao methods allowed in `main`.
+- Do not forget about Dependency Injection. Use your annotations.
+
+__You can check yourself using this__ [checklist](https://mate-academy.github.io/jv-program-common-mistakes/java-JDBC/dao-vs-service/dao-vs-service_checklist.html)
```
Comment (RomanDubovskyi) [inline README.md:29]: checklist linke doesn't work, i guess you're planning to add after

File: src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+import mate.jdbc.lib.exception.DataProcessingException;
+import mate.jdbc.model.Manufacturer;
+import mate.jdbc.util.ConnectionUtil;
+
+@Dao
+public class ManufacturerDaoImpl implements ManufacturerDao {
+    @Override
+    public Manufacturer create(Manufacturer manufacturer) {
+        String query = "INSERT INTO manufacturers (manufacturer_name,manufacturer_country) "
+                + "VALUES (?,?)";
```
Comment (RomanDubovskyi) [inline src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java:21]:
```suggestion
        String query = "INSERT INTO manufacturers (manufacturer_name, manufacturer_country) "
                + "VALUES (?, ?)";
```

File: src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+                    throwable);
+        }
+    }
+
+    @Override
+    public Optional<Manufacturer> get(Long id) {
+        String query = "SELECT * FROM manufacturers"
+                + " WHERE manufacturer_id = " + id + " AND deleted = FALSE";
+        try (Connection connection = ConnectionUtil.getConnection();
+                Statement statement = connection.createStatement()) {
```
Comment (RomanDubovskyi) [inline src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java:44]: why not prepared statement?

File: src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+            throw new DataProcessingException("Couldn't get manufacturer by id " + id + " ",
+                    throwable);
+        }
+    }
+
+    @Override
+    public List<Manufacturer> getAll() {
+        String query = "SELECT * FROM manufacturers WHERE deleted = FALSE";
+        try (Connection connection = ConnectionUtil.getConnection();
+                Statement statement = connection.createStatement()) {
```
Comment (RomanDubovskyi) [inline src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java:62]: What about this one?

**Review by RomanDubovskyi — APPROVED** [review]

File: src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+                PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            preparedStatement.setLong(1, id);
+            return preparedStatement.executeUpdate() > 0;
+        } catch (SQLException throwable) {
+            throw new DataProcessingException("Couldn't delete a manufacturer by id " + id + " ",
+                    throwable);
+        }
+    }
+
+    private Manufacturer setManufacturer(ResultSet resultSet) throws SQLException {
```
Comment (boroda4436) [inline src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java:107]:
```suggestion
    private Manufacturer getManufacturer(ResultSet resultSet) throws SQLException {
```

File: README.md [hunk 10]
```diff
@@ -1 +1,29 @@
...
+    - Driver create(Driver driver);
+    - Driver get(Long id);
+    - List<Driver> getAll();
+    - Driver update(Driver driver);
+    - boolean delete(Long id);
+
+- Test your code in `main`.
+- Main should only contain invocation of `Service` methods, no Dao methods allowed in `main`.
+- Do not forget about Dependency Injection. Use your annotations.
+
```
Comment (boroda4436) [inline README.md:28]: - Add required dependencies to the `pom.xml` file

**Review by boroda4436 — APPROVED** [review]

**Pull request merged**